extensions/gitpod/src/remoteConnector.ts

```diff
@@ -379,63 +380,149 @@ export default class LocalApp extends Disposable {
 		}
 	}
 
+	private async getWorkspaceSSHDestination(workspaceId: string): Promise<string | undefined> {
+		const session = await vscode.authentication.getSession(
+			'gitpod',
+			['function:getWorkspace', 'function:getOwnerToken', 'function:getLoggedInUser', 'resource:default'],
+			{ createIfNone: true }
+		);
+
+		const gitpodHost = vscode.workspace.getConfiguration('gitpod').get<string>('host')!;
+		const serviceUrl = new URL(gitpodHost);
+
+		const workspaceInfo = await withServerApi(session.accessToken, serviceUrl.toString(), service => service.server.getWorkspace(workspaceId), this.logger);
+		const workspaceUrl = new URL(workspaceInfo.latestInstance!.ideUrl);
+
+		const sshHostKeyEndPoint = `https://${workspaceUrl.host}/_ssh/host_keys`;
+		const sshHostKeyResponse = await fetch(sshHostKeyEndPoint);
+		if (!sshHostKeyResponse.ok) {
+			// Gitpod SSH gateway not configured
+			return;
+		}
+
+		const ownerToken = await withServerApi(session.accessToken, serviceUrl.toString(), service => service.server.getOwnerToken(workspaceId), this.logger);
```

Comment:
Should not we rather minimize calls to `withServerApi` it establishes persistent connection.

Reply:
I think they are kinda expensive as it creates a ws connection but it's fine as long as we don't create hundreds of them and `withServerApi` closes the connection immediately when returning

Reply:
it is just we do it twice in this method, maybe we can group it?

Reply:
actually let's do like that we are most likely to switch to Public API next week already and when it is not important